lms/djangoapps/grades/new/course_grade.py

```diff
@@ -0,0 +1,205 @@
+"""
+CourseGrade Class
+"""
+
+from collections import defaultdict
+from django.conf import settings
+from lazy import lazy
+from logging import getLogger
+from course_blocks.api import get_course_blocks
+from openedx.core.djangoapps.signals.signals import GRADES_UPDATED
+from xmodule import block_metadata_utils
+
+from .subsection_grade import SubsectionGradeFactory
+
+
+log = getLogger(__name__)
+
+
+class CourseGrade(object):
+    """
+    Course Grade class
+    """
+    def __init__(self, student, course, course_structure):
+        self.student = student
+        self.course = course
+        self.course_structure = course_structure
+        self.chapter_grades = []
+
+    @lazy
+    def subsection_grade_totals_by_format(self):
+        """
+        Returns grades for the subsections in the course in
+        a dict keyed by subsection format types.
+        """
+        subsections_by_format = defaultdict(list)
+        for chapter in self.chapter_grades:
+            for subsection_grade in chapter['sections']:
+                if subsection_grade.graded:
+                    graded_total = subsection_grade.graded_total
+                    if graded_total.possible > 0:
+                        subsections_by_format[subsection_grade.format].append(graded_total)
+        return subsections_by_format
+
+    @lazy
+    def locations_to_scores(self):
+        """
+        Returns a dict of problem scores keyed by their locations.
+        """
+        locations_to_scores = {}
+        for chapter in self.chapter_grades:
+            for subsection_grade in chapter['sections']:
+                locations_to_scores.update(subsection_grade.locations_to_scores)
+        return locations_to_scores
+
+    @property
+    def has_access_to_course(self):
+        """
+        Returns whether the course structure as seen by the
+        given student is non-empty.
+        """
+        return len(self.course_structure) > 0
+
+    @lazy
+    def summary(self):
+        """
+        Returns the grade summary as calculated by the course's grader.
+        """
+        # Grading policy might be overriden by a CCX, need to reset it
+        self.course.set_grading_policy(self.course.grading_policy)
+        grade_summary = self.course.grader.grade(
+            self.subsection_grade_totals_by_format,
+            generate_random_scores=settings.GENERATE_PROFILE_SCORES
+        )
+
+        # We round the grade here, to make sure that the grade is a whole percentage and
+        # doesn't get displayed differently than it gets grades
+        grade_summary['percent'] = round(grade_summary['percent'] * 100 + 0.05) / 100
```

Comment:
I believe this is it's way of rounding up.  Pre-existing logic which I hadn't touched.  If we do revisit this (in a subsequent phase), it may need to be backward compatible.